edgeql: Augment DELETE syntax. (#469)
edgedb/edgedb: docs/edgeql/statements/delete.rst

.. eql:synopsis::

    [ WITH <with-item> [, ...] ]

    DELETE <expr>

    [ FILTER <filter-expr> ]

    [ ORDER BY <order-expr> [direction] [THEN ...] ]

    [ OFFSET <offset-expr> ]

    [ LIMIT  <limit-expr> ] ;

:eql:synopsis:`WITH`
    Alias declarations.

    The ``WITH`` clause allows specifying module aliases as well
    as expression aliases that can be referenced by the ``DELETE``
    statement.  See :ref:`ref_eql_with` for more information.

:eql:synopsis:`DELETE <expr>`

>     In principle, the entire ``DELETE ...`` statement is syntactic

Reviewer: ```suggestion
    The :eql:synopsis:`DELETE <expr>` statement is syntactic sugar for 
```

Author: OK, I'll update the synopsis section to `DELETE ...` instead.

    statement.  See :ref:`ref_eql_with` for more information.

:eql:synopsis:`DELETE <expr>`
    Form the base of the set to be deleted. This base set can be
    further trimmed with other clauses.

:eql:synopsis:`FILTER <filter-expr>`
    The optional ``FILTER`` clause, where :eql:synopsis:`<filter-expr>`
    is any expression that has a result of type :eql:type:`bool`. The
    condition is evaluated for every element in the base set produced

>     by the ``DELETE`` clause.  The result of the evaluation of the

Reviewer: > produced
     by the ``DELETE`` clause

This is ambiguous. 

I guess you've just copy/pasted docs from the SELECT statement. We don't want that kind of duplication. I suggest to do it another way: copy/paste the synopsis and simply refer to the SELECT statement for the explanation of how different clauses work.  First paragraph should mention that `DELETE ...` is a sugar for `DELETE (SELECT ...)`; that should be enough.

Author: I've reworded that section.
